### PR TITLE
Move all `requirements` to `pyproject.toml`

### PR DESCRIPTION
**Pull request by ArthurConmy**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,14 @@ furo = {version = ">=2022.12.7", optional = true, python = ">=3.8,<3.10"}
...
 tabulate= {version = ">=0.9.0", optional = true, python = ">=3.8,<3.10"}
 typeguard = "^3.0.2"
+ipython = "^8.14.0"
+plotly = "^5.15.0"
+circuitsvis = "^1.40.1"
+streamlit = "^1.25.0"
+nltk = "^3.8.1"
+
+# pattern has a stupid dependency on mysqllib which requires extra linux package installs so breaks streamlit. This guy's fork is an open PR that avoids this crap <3
+pattern = {git = "https://github.com/uob-vil/pattern.git", rev = "3b77852c51d1e80b17d9abba0be5efad6a2fd7a5"}
```
Comment (ArthurConmy) [inline pyproject.toml:38]: Sneaky way to deal with `pattern` crap

File: explore_prompts/requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# TODO remove all this, probably along with all of repo/expore_prompts
```
Comment (ArthurConmy) [inline explore_prompts/requirements.txt:1]: @callummcdougall I don't think I need this explore_prompts directory at all. Can we delete?

File: pyproject.toml (hunk 1 above)
Comment (callummcdougall) [reply pyproject.toml:38]: lgtm!

File: explore_prompts/requirements.txt (hunk 2 above)
Comment (callummcdougall) [reply explore_prompts/requirements.txt:1]: Yep, if you don't need anything from it, it can be fully deleted

**Pull request merged**